readthedocs/rtd_tests/tests/test_redirects.py

```diff
@@ -215,6 +215,36 @@ def test_redirect_htmldir(self):
             r['Location'], 'http://pip.readthedocs.org/en/latest/faq/')
 
 
+class CustomRedirectTests(TestCase):
```

Comment:
Most of the redirect tests can be written without using a client so instead of refactoring everything I just added a new test case.